lib/fluent/supervisor.rb

```diff
@@ -268,7 +268,8 @@ def initialize(path, level, chuser, chgroup, opts)
 
       def init
         if @path && @path != "-"
-          @io = File.open(@path, "a")
+          Dir.mkdir(File.dirname(@path)) unless File.exists?(@path)
```

Comment:
`File.exists?` (and `Dir.exists?`) are deprecated methods. Use `File.exist?` instead.